generate_secrets.py

```diff
@@ -23,9 +23,11 @@ def main():
     if not os.path.exists(settings.GPG_KEY_PASSWORD_FILENAME):
         gpg_key_password = get_random_secret_key()
         gpg = gnupg.GPG(gnupghome=settings.SECRETS_DIR)
+
         input_data = gpg.gen_key_input(
             name_email=settings.EMAIL_FROM_ADDR,
-            passphrase=gpg_key_password
+            passphrase=gpg_key_password,
+            expect_passphrase=False
```

Comment:
Could you explain what this option does, what issue you ran into, and why this solves the issue? Thanks!

Reply:
@james-perretta Absolutely! I actually created an issue for this but possibly in the wrong repo: https://github.com/eecs-autograder/autograder-full-stack/issues/9

When I was following the setup instructions, I couldn't get execution past this line for generating a key. It was hanging inside of this call to gen_key_input. It seems like there are a ton of ways to set up pinentry / other forms of handling requesting passphrase from the user, but in this case you don't want the user to enter a passphrase (since its being passed in as an argument).

After poking at that line for a bit, I realized it was still expecting a passphrase from the user despite the fact that we're giving it one as an arg. Somehow passing this as false disabled this behavior.

The more appealing option was setting pinentry mode to loopback, but that didn't appear to work.

The way I found this option was seeing it mentioned throughout the documentation for other methods that `gpg` provides: https://gnupg.readthedocs.io/en/latest/

It claims that `gen_key_input` just converts the passed in args to command line arguments, but I believe they may have just forgotten to document this new arg since it does seem to change the behavior.